### PR TITLE
[REF] payment(_stripe): cleanup express checkout implementation

### PR DESCRIPTION
**Pull request by chevalierv**
Fine-tuning of https://github.com/odoo/odoo/commit/c624ed56038db524d781df4e034f0fe501e777d9.

task-3004118

### CODE REVIEW

File: addons/payment_stripe/static/src/js/express_checkout_form.js [hunk 1]
```diff
@@ -20,53 +20,42 @@ paymentExpressCheckoutForm.include({
...
-                pending: isShippingInformationRequired,
+        const paymentRequest = stripeJS.paymentRequest({
+            ...{
+                country: providerData.countryCode,
+                currency: this.txContext.currencyName,
+                requestPayerName: true, // Force fetching the billing address for Apple Pay.
+                requestPayerEmail: true,
+                requestPayerPhone: true,
+                requestShipping: this._isShippingInformationRequired(),
             },
```
Comment (Williambraecky) [inline addons/payment_stripe/static/src/js/express_checkout_form.js:34]: You don't need to unpack an object if you are creating it right then and there:
```suggestion
            country: providerData.countryCode,
            currency: this.txContext.currencyName,
            requestPayerName: true, // Force fetching the billing address for Apple Pay.
            requestPayerEmail: true,
            requestPayerPhone: true,
            requestShipping: this._isShippingInformationRequired(),
```

File: addons/payment_stripe/static/src/js/express_checkout_form.js [hunk 2]
```diff
@@ -80,21 +69,19 @@ paymentExpressCheckoutForm.include({
...
+                    name: ev.shippingAddress.recipient,
+                    phone: ev.shippingAddress.phone,
+                    street: ev.shippingAddress.addressLine[0],
+                    street2: ev.shippingAddress.addressLine[1],
+                    zip: ev.shippingAddress.postalCode,
+                    city: ev.shippingAddress.city,
+                    country: ev.shippingAddress.country,
+                    state: ev.shippingAddress.region,
+                }
+                addresses.shipping_option = ev.shippingOption
```
Comment (Williambraecky) [inline addons/payment_stripe/static/src/js/express_checkout_form.js:83]:
```suggestion
                addresses.shipping_address = {
                    name: ev.shippingAddress.recipient,
                    phone: ev.shippingAddress.phone,
                    street: ev.shippingAddress.addressLine[0],
                    street2: ev.shippingAddress.addressLine[1],
                    zip: ev.shippingAddress.postalCode,
                    city: ev.shippingAddress.city,
                    country: ev.shippingAddress.country,
                    state: ev.shippingAddress.region,
                };
                addresses.shipping_option = ev.shippingOption;
```

File: addons/payment_stripe/static/src/js/express_checkout_form.js [hunk 3]
```diff
@@ -151,51 +132,27 @@ paymentExpressCheckoutForm.include({
...
-                            pending: false,
+                        ...{
+                            status: 'success',
+                            shippingOptions: availableCarriers.map(carrier => ({
+                                id: String(carrier.id),
+                                label: carrier.name,
+                                detail: carrier.description ? carrier.description:"",
+                                amount: carrier.minorAmount,
+                            })),
                         },
```
Comment (Williambraecky) [inline addons/payment_stripe/static/src/js/express_checkout_form.js:143]: Same here no need to unpack a new object

File: addons/payment_stripe/static/src/js/express_checkout_form.js [hunk 4]
```diff
@@ -151,51 +132,27 @@ paymentExpressCheckoutForm.include({
...
+            paymentRequest.on('shippingoptionchange', async (ev) => {
                 ev.updateWith({
-                    status: 'success',
-                    total: {
-                        label: this.txContext.merchantName,
-                        amount: this.txContext.minorAmount + ev.shippingOption.amount,
-                        pending: false,
+                    ...{
+                        status: 'success',
                     },
```
Comment (Williambraecky) [inline addons/payment_stripe/static/src/js/express_checkout_form.js:154]: Same here

**Review by Williambraecky — COMMENTED** [review]
Besides those points (which are details) it looks ok to me :)

**Review by AntoineVDV — APPROVED** [review]
FYI The diff in `website_sale` is a cleanup of what was done in https://github.com/odoo/odoo/pull/88374 and of the `transaction_id` route param that was actually never passed since at least 14.3.

@robodoo r+

**Pull request closed without merging**